File: app/src/components/MacRadarPaneli.js
// ---------------------------------------------------------------------------
// MAÇ RADAR PANELİ — Radar 3 / 4 / 5, YALNIZ BU MAÇ İÇİN (2026-08-07)
// ---------------------------------------------------------------------------
// KULLANICI İSTEĞİ: "Radar 3-4-5'i maç detayına koyalım ama her maçın KENDİ
// sırasını koyacağız." Radar sekmesinde bu üç panel bültenin 15 maçını birden
// listeliyor; maçın içindeyken o kalabalığa gerek yok — tek satır lazım.
//
// NE GÖSTERİR (hepsi mevcut uçlardan, yeni uç açılmadı):
//   • Radar 3 — Oynanma DNA: bu maçın günlük 1/X/2 oynanma yüzdeleri, her
//     kaynak ayrı satır. Kaynak satırına dokununca geçmiş DNA paneli açılır.
//   • Radar 4 — Oran Takibi: bu maçın seçili gündeki mühürlü 1/X/2 oranı ve
//     bir önceki güne göre yönü.
//   • Radar 5 — Bülten DNA: bu maçın BÜLTEN SIRASININ geçmişi (aynı sırada
//     oynanmış maçlar, sonuçlarıyla).
//
// TASARIM KARARI: satırları Radar ekranındaki bileşenlerin AYNISI çizer
// (`RadarDayRows`, `SiraGecmisListesi`). Kopya bileşen yazılmadı; iki ekran
// aynı veriyi farklı gösterirse hangisinin doğru olduğu bilinemez.
//
// DÜRÜSTLÜK: veri yoksa uydurma yüzde/oran yazılmaz; ilgili bölüm sebebini
// yazar. Gelecek gün seçilirse yüzde basılmaz ("bu gün henüz gelmedi").

import React, { useEffect, useState } from 'react';
import {
  View, Text, StyleSheet, ActivityIndicator, TouchableOpacity, ScrollView,
} from 'react-native';

import { api } from '../api';
import { colors, spacing, radius } from '../theme';
import { MarketRow, PublicRow } from './RadarDayRows';
import SiraGecmisListesi from './SiraGecmisListesi';

const KISA_GUN = {
  Pazartesi: 'Pzt', Salı: 'Sal', Çarşamba: 'Çar', Perşembe: 'Per',
  Cuma: 'Cum', Cumartesi: 'Cmt', Pazar: 'Paz',
};

/** Gün seçici — Radar ekranındakiyle aynı mantık, dar alana göre kısaltılmış. */
function GunSecici({ gunler, secili, onSec }) {
  if (!gunler?.length) return null;
  return (
    <ScrollView horizontal showsHorizontalScrollIndicator={false} contentContainerStyle={st.gunSira}>
      {gunler.map((g) => {
        const acik = g.date === secili;
        return (
          <TouchableOpacity
            key={g.date}
            style={[st.gun, acik && st.gunAcik, g.future && st.gunGelecek]}
            onPress={() => onSec(g.date)}
            accessibilityRole="button"
            accessibilityLabel={`${g.weekday || g.date} gününü seç`}
          >
            <Text style={[st.gunTxt, acik && st.gunTxtAcik]}>
              {KISA_GUN[g.weekday] || g.weekday || g.date}
            </Text>
            <Text style={[st.gunAlt, acik && st.gunTxtAcik]}>{(g.date || '').slice(5)}</Text>
          </TouchableOpacity>
        );
      })}
    </ScrollView>
  );
}

function Bolum({ baslik, alt, children }) {
  return (
    <View style={st.kart}>
      <Text style={st.baslik}>{baslik}</Text>
      {alt ? <Text style={st.altBaslik}>{alt}</Text> : null}
      {children}
    </View>
  );
}

/**
 * Son (en güncel) veri günü: gelecek olmayan günlerin sonuncusu.
 * Hiç uygun gün yoksa null döner — varsayılan gün UYDURULMAZ.
 */
function sonGun(gunler) {
  const uygun = (gunler || []).filter((g) => !g.future);
  return uygun.length ? uygun[uygun.length - 1].date : ((gunler || [])[0]?.date ?? null);
}

export default function MacRadarPaneli({ m }) {
  const roundId = m?.roundId ?? null;
  const no = m?.no ?? null;

  const [oynanma, setOynanma] = useState({ yukleniyor: true, veri: null, hata: null });
  const [oran, setOran] = useState({ yukleniyor: true, veri: null, hata: null });
  const [siraDna, setSiraDna] = useState({ yukleniyor: true, veri: null, hata: null });
  const [siraMaclari, setSiraMaclari] = useState({ yukleniyor: true, liste: [], hata: null });

  const [oynanmaGun, setOynanmaGun] = useState(null);
  const [oranGun, setOranGun] = useState(null);
  const [acikDna, setAcikDna] = useState(null);   // "<no>|<kaynak>"

  // Radar 3 — günlük oynanma yüzdeleri (yalnız bu maç).
  useEffect(() => {
    let iptal = false;
    if (roundId == null || no == null) return undefined;
    api.radarDailyPlayed(roundId, no)
      .then((d) => {
        if (iptal) return;
        setOynanma({ yukleniyor: false, veri: d, hata: null });
        setOynanmaGun((x) => x || sonGun(d?.days));
      })
      .catch((e) => { if (!iptal) setOynanma({ yukleniyor: false, veri: null, hata: e?.message || 'okunamadı' }); });
    return () => { iptal = true; };
  }, [roundId, no]);

  // Radar 4 — günlük mühürlü oranlar (yalnız bu maç).
  useEffect(() => {
    let iptal = false;
    if (roundId == null || no == null) return undefined;
    api.radarDailyOdds(roundId, no)
      .then((d) => {
        if (iptal) return;
        setOran({ yukleniyor: false, veri: d, hata: null });
        setOranGun((x) => x || sonGun(d?.days));
      })
      .catch((e) => { if (!iptal) setOran({ yukleniyor: false, veri: null, hata: e?.message || 'okunamadı' }); });
    return () => { iptal = true; };
  }, [roundId, no]);

  // Radar 5 — bu SIRANIN bülten DNA'sı + aynı sırada oynanmış geçmiş maçlar.
  useEffect(() => {
    let iptal = false;
    if (roundId == null || no == null) return undefined;
    api.radarPositionDna(roundId)
      .then((d) => { if (!iptal) setSiraDna({ yukleniyor: false, veri: d, hata: null }); })
      .catch((e) => { if (!iptal) setSiraDna({ yukleniyor: false, veri: null, hata: e?.message || 'okunamadı' }); });
    api.radarPositionMatches(no, roundId)
      .then((d) => { if (!iptal) setSiraMaclari({ yukleniyor: false, liste: d?.matches || [], hata: null }); })
      .catch((e) => { if (!iptal) setSiraMaclari({ yukleniyor: false, liste: [], hata: e?.message || 'okunamadı' }); });
    return () => { iptal = true; };
  }, [roundId, no]);

  if (roundId == null || no == null) {
    return (
      <View style={st.kart}>
        <Text style={st.bos}>Bu maç için hafta/sıra bilgisi yok — radar gösterilemez.</Text>
      </View>
    );
  }

  // Satır bileşenlerinin beklediği asgari şekil: { no, home, away }.
  const satir = {
    no,
    home: m?.home?.mediumName || m?.home?.name || 'Ev sahibi',
    away: m?.away?.mediumName || m?.away?.name || 'Deplasman',
  };

  // Radar 5: bu sıranın geçmiş dağılımı (varsa).
  const siraKaydi = (siraDna.veri?.dna?.positions || []).find((p) => Number(p.position) === Number(no)) || null;

  return (
    <View>
      {/* ——— RADAR 3 ——— */}
      <Bolum
        baslik="Radar 3 · Oynanma DNA"
        alt={`${no}. sıra — günlük 1/X/2 oynanma yüzdeleri. Her kaynak ayrı satır; dokununca o kaynağın geçmişi açılır.`}
      >
        {oynanma.yukleniyor ? (
          <ActivityIndicator color={colors.primary} style={st.bekle} />
        ) : oynanma.hata ? (
          <Text style={st.bos}>Oynanma verisi okunamadı: {oynanma.hata}</Text>
        ) : (
          <>
            <GunSecici gunler={oynanma.veri?.days} secili={oynanmaGun} onSec={setOynanmaGun} />
            <PublicRow
              item={satir}
              data={oynanma.veri}
              day={oynanmaGun}
              openKey={acikDna}
              onToggleDna={setAcikDna}
              roundId={roundId}
              tick={0}
            />
          </>
        )}
      </Bolum>

      {/* ——— RADAR 4 ——— */}
      <Bolum
        baslik="Radar 4 · Oran Takibi"
        alt={`${no}. sıra — maç öncesi mühürlenmiş günlük 1/X/2 oranı ve bir önceki güne göre yönü.`}
      >
        {oran.yukleniyor ? (
          <ActivityIndicator color={colors.primary} style={st.bekle} />
        ) : oran.hata ? (
          <Text style={st.bos}>Oran verisi okunamadı: {oran.hata}</Text>
        ) : (
          <>
            <GunSecici gunler={oran.veri?.days} secili={oranGun} onSec={setOranGun} />
            <MarketRow item={satir} data={oran.veri} day={oranGun} />
          </>
        )}
      </Bolum>

      {/* ——— RADAR 5 ——— */}
      <Bolum
        baslik="Radar 5 · Bülten DNA"
        alt={`${no}. sırada geçmişte oynanmış maçlar ve sonuçları. Bu maça değil, BU SIRAYA bakar.`}
      >
        {siraDna.yukleniyor ? (
          <ActivityIndicator color={colors.primary} style={st.bekle} />
        ) : siraDna.hata ? (
          <Text style={st.bos}>Sıra DNA'sı okunamadı: {siraDna.hata}</Text>
        ) : (
          <>
            {(() => {
              // Dağılım TÜM ZAMANLAR penceresinden okunur; örneklem yoksa
              // yüzde YAZILMAZ (0 göstermek "hiç olmamış" demek olurdu).
              const w = siraKaydi?.windows?.allTime || null;
              if (!w || !w.sample) {
                return <Text style={st.bos}>Bu sıra için doğrulanmış geçmiş dağılım yok.</Text>;
              }
              return (
                <View style={st.dagilim}>
                  {['1', 'X', '2'].map((k) => (
                    <View key={k} style={st.dagilimHucre}>
                      <Text style={st.dagilimK}>{k}</Text>
                      <Text style={st.dagilimV}>{w.pct?.[k] == null ? '–' : `%${w.pct[k]}`}</Text>
                    </View>
                  ))}
                  <Text style={st.dagilimNot}>
                    {`${w.sample} sonuçlanmış maç`}
                    {siraKaydi.sampleLabel ? ` · ${siraKaydi.sampleLabel}` : ''}
                  </Text>
                </View>
              );
            })()}
            <SiraGecmisListesi kayit={siraMaclari} />
          </>
        )}
      </Bolum>
    </View>
  );
}

const st = StyleSheet.create({
  kart: {
    backgroundColor: colors.surface, borderRadius: radius.md, padding: spacing.md,
    marginBottom: spacing.md, borderWidth: 1, borderColor: colors.border,
  },
  baslik: { color: colors.text, fontSize: 13.5, fontWeight: '900' },
  altBaslik: { color: colors.textMuted, fontSize: 10.5, lineHeight: 14, marginTop: 2, marginBottom: 8 },
  bos: { color: colors.textMuted, fontSize: 11.5, lineHeight: 16 },
  bekle: { marginVertical: 12 },

  gunSira: { gap: 5, paddingBottom: 8 },
  gun: {
    paddingHorizontal: 9, paddingVertical: 5, borderRadius: radius.sm,
    backgroundColor: colors.bgAlt, alignItems: 'center', minWidth: 46,
  },
  gunAcik: { backgroundColor: colors.primary },
  gunGelecek: { opacity: 0.55 },
  gunTxt: { color: colors.textSoft, fontSize: 11, fontWeight: '800' },
  gunTxtAcik: { color: colors.white },
  gunAlt: { color: colors.textMuted, fontSize: 9 },

  dagilim: { flexDirection: 'row', alignItems: 'center', gap: 8, marginBottom: 8, flexWrap: 'wrap' },
  dagilimHucre: {
    backgroundColor: colors.bgAlt, borderRadius: radius.sm,
    paddingHorizontal: 9, paddingVertical: 4, alignItems: 'center', minWidth: 52,
  },
  dagilimK: { color: colors.textMuted, fontSize: 9.5, fontWeight: '900' },
  dagilimV: { color: colors.text, fontSize: 13, fontWeight: '900' },
  dagilimNot: { color: colors.textMuted, fontSize: 10 },
});

File: flutter/lib/features/match_detail/mac_radar_paneli.dart
// KAYNAK: app/src/components/MacRadarPaneli.js — BİREBİR çeviri.
//
// MAÇ RADAR PANELİ — Radar 3 / 4, YALNIZ BU MAÇ İÇİN (2026-08-07)
//
// KULLANICI İSTEĞİ: "Radar 3-4-5'i maç detayına koyalım ama her maçın KENDİ
// sırasını koyacağız." Radar sekmesinde bu paneller bültenin 15 maçını birden
// listeliyor; maçın içindeyken o kalabalığa gerek yok — tek satır lazım.
//
// RADAR 5 KALDIRILDI (2026-08-08): kullanıcı "radar5 bugün yaptığımızı
// beğenmedim" dedi. Bülten DNA bölümü buradan çıkarıldı.
//
// TASARIM KARARI: satırları Radar ekranındaki bileşenlerin AYNISI çizer
// (`radar_day_rows.dart`). Kopya bileşen yazılmadı; iki ekran aynı veriyi
// farklı gösterirse hangisinin doğru olduğu bilinemez.
//
// ═══════════════ PARAMETRE TUZAĞI (2026-08-07 hatası) ══════════════════════
// `radarDailyPlayed(roundId, matchId)` ve `radarDailyOdds(roundId, matchId)`
// ikinci parametreyi `matchId` olarak alır ve backend bununla GÖZLEMLERİ
// süzer. `matchId` BÜLTEN SIRASI DEĞİLDİR — Spor Toto maç kimliğidir. İlk
// sürümde buraya sıra numarası (1) geçilmişti; hiçbir gözlemle eşleşmedi,
// hücreler boş döndü ve ekran "bu gün için kayıt yok" yazdı. Veri vardı, sorgu
// yanlıştı — ve bu SESSİZ bir hataydı: uç 200 döndü, hata basılmadı, yalnız
// sayılar yoktu.
//
// KURAL: bu iki uç haftanın TAMAMI için çağrılır; maç süzmesi ekranda
// `item.no` ile yapılır. Aşağıdaki çağrılarda matchId GEÇİLMEZ.
//
// İKİ AYRI SEKME (kullanıcı isteği, 2026-08-11): panel artık "Oynanma
// Yüzdeleri" ve "Oran Takibi" sekmelerinde AYRI AYRI çiziliyor. Kartlar
// kopyalanmadı; hangi bölümün çizileceğini `bolum` söyler. Varsayılan
// `hepsi` — parametresiz çağıran eski kullanım aynen çalışır.

import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../core/network/api_client.dart';
import '../../core/theme/tokens.dart';
import '../radar/played_dna_panel.dart';
import '../radar/radar_day_rows.dart';

const Map<String, String> _kisaGun = {
  'Pazartesi': 'Pzt',
  'Salı': 'Sal',
  'Çarşamba': 'Çar',
  'Perşembe': 'Per',
  'Cuma': 'Cum',
  'Cumartesi': 'Cmt',
  'Pazar': 'Paz',
};

/// matchId GEÇİLMEZ — bkz. dosya başındaki "PARAMETRE TUZAĞI" notu.
final _dailyPlayedProvider = FutureProvider.autoDispose
    .family<Map<String, dynamic>, Object>((ref, roundId) async {
      final d = await api.radarDailyPlayed(roundId);
      return Map<String, dynamic>.from(d as Map);
    });

final _dailyOddsProvider = FutureProvider.autoDispose
    .family<Map<String, dynamic>, Object>((ref, roundId) async {
      final d = await api.radarDailyOdds(roundId);
      return Map<String, dynamic>.from(d as Map);
    });

/// Son (en güncel) veri günü: gelecek OLMAYAN günlerin sonuncusu.
/// Hiç uygun gün yoksa ilk gün — varsayılan gün UYDURULMAZ.
String? _sonGun(List? gunler) {
  final uygun = (gunler ?? const [])
      .cast<Map>()
      .where((g) => g['future'] != true)
      .toList();
  if (uygun.isNotEmpty) return uygun.last['date'] as String?;
  final hepsi = (gunler ?? const []).cast<Map>();
  return hepsi.isEmpty ? null : hepsi.first['date'] as String?;
}

/// Panelin hangi bölümü çizilsin? `hepsi` iki kartı da çizer (eski davranış).
enum MacRadarBolumu { hepsi, oynanma, oran }

class MacRadarPaneli extends ConsumerStatefulWidget {
  const MacRadarPaneli({
    super.key,
    required this.m,
    this.bolum = MacRadarBolumu.hepsi,
  });

  final Map<String, dynamic> m;
  final MacRadarBolumu bolum;

  @override
  ConsumerState<MacRadarPaneli> createState() => _MacRadarPaneliState();
}

class _MacRadarPaneliState extends ConsumerState<MacRadarPaneli> {
  String? _oynanmaGun;
  String? _oranGun;
  String? _acikDna; // "<no>|<kaynak>"

  @override
  Widget build(BuildContext context) {
    final roundId = widget.m['roundId'];
    final no = widget.m['no'];

    if (roundId == null || no == null) {
      return MacRadarKart(
        children: [
          Text(
            'Bu maç için hafta/sıra bilgisi yok — radar gösterilemez.',
            style: kMacRadarBos,
          ),
        ],
      );
    }

    // Satır bileşenlerinin beklediği asgari şekil: { no, home, away }.
    final home = widget.m['home'] as Map?;
    final away = widget.m['away'] as Map?;
    final satir = {
      'no': no,
      'home': home?['mediumName'] ?? home?['name'] ?? 'Ev sahibi',
      'away': away?['mediumName'] ?? away?['name'] ?? 'Deplasman',
    };

    // Sekme yalnız bir bölümü gösteriyorsa ÖTEKİNİN İSTEĞİ HİÇ ATILMAZ:
    // kullanıcı Oran Takibi'ne bakarken oynanma verisi çekmek boşuna trafiktir.
    final oynanmaVar = widget.bolum != MacRadarBolumu.oran;
    final oranVar = widget.bolum != MacRadarBolumu.oynanma;
    final oynanma = oynanmaVar
        ? ref.watch(_dailyPlayedProvider(roundId))
        : null;
    final oran = oranVar ? ref.watch(_dailyOddsProvider(roundId)) : null;

    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      children: [
        // ——— RADAR 3 ———
        if (oynanmaVar)
          MacRadarKart(
            baslik: 'Radar 3 · Oynanma DNA',
            alt:
                '$no. sıra — günlük 1/X/2 oynanma yüzdeleri. Her kaynak ayrı '
                'satır; dokununca o kaynağın geçmişi açılır.',
            children: [
              oynanma!.when(
                loading: () => const MacRadarBekle(),
                error: (e, _) =>
                    Text('Oynanma verisi okunamadı: $e', style: kMacRadarBos),
                data: (d) {
                  final secili = _oynanmaGun ?? _sonGun(d['days'] as List?);
                  return Column(
                    crossAxisAlignment: CrossAxisAlignment.stretch,
                    children: [
                      _GunSecici(
                        gunler: d['days'] as List?,
                        secili: secili,
                        onSec: (g) => setState(() => _oynanmaGun = g),
                      ),
                      PublicRow(
                        item: satir,
                        data: d,
                        day: secili,
                        openKey: _acikDna,
                        onToggleDna: (k) => setState(() => _acikDna = k),
                        dnaPanelBuilder: (kaynak) => PlayedDnaPanel(
                          roundId: roundId,
                          no: no,
                          source: kaynak,
                          day: secili,
                          // Maç detayında otomatik tazeleme yok (Radar ekranının
                          // 60 sn'lik sayacı buraya taşınmadı — kaynakta da 0).
                          tick: 0,
                        ),
                      ),
                    ],
                  );
                },
              ),
            ],
          ),

        // ——— RADAR 4 ———
        if (oranVar)
          MacRadarKart(
            baslik: 'Radar 4 · Oran Takibi',
            alt:
                '$no. sıra — maç öncesi mühürlenmiş günlük 1/X/2 oranı ve bir '
                'önceki güne göre yönü.',
            children: [
              oran!.when(
                loading: () => const MacRadarBekle(),
                error: (e, _) =>
                    Text('Oran verisi okunamadı: $e', style: kMacRadarBos),
                data: (d) {
                  final secili = _oranGun ?? _sonGun(d['days'] as List?);
                  return Column(
                    crossAxisAlignment: CrossAxisAlignment.stretch,
                    children: [
                      _GunSecici(
                        gunler: d['days'] as List?,
                        secili: secili,
                        onSec: (g) => setState(() => _oranGun = g),
                      ),
                      MarketRow(item: satir, data: d, day: secili),
                    ],
                  );
                },
              ),
            ],
          ),
      ],
    );
  }
}

/// Gün seçici — Radar ekranındakiyle aynı mantık, dar alana göre kısaltılmış.
class _GunSecici extends StatelessWidget {
  const _GunSecici({
    required this.gunler,
    required this.secili,
    required this.onSec,
  });

  final List? gunler;
  final String? secili;
  final ValueChanged<String> onSec;

  @override
  Widget build(BuildContext context) {
    if (gunler == null || gunler!.isEmpty) return const SizedBox.shrink();
    return SingleChildScrollView(
      scrollDirection: Axis.horizontal,
      padding: const EdgeInsets.only(bottom: 8),
      child: Row(
        children: [
          for (final g in gunler!.cast<Map>()) ...[
            _gun(g),
            const SizedBox(width: 5),
          ],
        ],
      ),
    );
  }

  Widget _gun(Map g) {
    final acik = g['date'] == secili;
    final gelecek = g['future'] == true;
    final weekday = '${g['weekday'] ?? ''}';
    final date = '${g['date'] ?? ''}';

    return Semantics(
      button: true,
      label: '${weekday.isNotEmpty ? weekday : date} gününü seç',
      child: Opacity(
        opacity: gelecek ? 0.55 : 1,
        child: GestureDetector(
          onTap: () => onSec(date),
          behavior: HitTestBehavior.opaque,
          child: Container(
            constraints: const BoxConstraints(minWidth: 46),
            padding: const EdgeInsets.symmetric(horizontal: 9, vertical: 5),
            decoration: BoxDecoration(
              color: acik ? AppColors.primary : AppColors.bgAlt,
              borderRadius: AppRadius.smR,
            ),
            child: Column(
              mainAxisSize: MainAxisSize.min,
              children: [
                Text(
                  _kisaGun[weekday] ?? (weekday.isNotEmpty ? weekday : date),
                  style: TextStyle(
                    color: acik ? AppColors.onPrimary : AppColors.textSoft,
                    fontSize: 11,
                    fontWeight: AppFont.heavy,
                  ),
                ),
                Text(
                  date.length > 5 ? date.substring(5) : date,
                  style: TextStyle(
                    color: acik ? AppColors.onPrimary : AppColors.textMuted,
                    fontSize: 9,
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

/// Maç detayındaki radar kartlarının ortak kabuğu. Bülten Sırası paneli de
/// bunu kullanır — kopya kart yazılmadı ki üç sekme aynı çerçevede görünsün.
class MacRadarKart extends StatelessWidget {
  const MacRadarKart({
    super.key,
    this.baslik,
    this.alt,
    required this.children,
  });

  final String? baslik;
  final String? alt;
  final List<Widget> children;

  @override
  Widget build(BuildContext context) => Container(
    margin: const EdgeInsets.only(bottom: Spacing.md),
    padding: const EdgeInsets.all(Spacing.md),
    decoration: BoxDecoration(
      color: AppColors.surface,
      borderRadius: AppRadius.mdR,
      border: Border.all(color: AppColors.border),
    ),
    child: Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      children: [
        if (baslik != null)
          Text(
            baslik!,
            style: TextStyle(
              color: AppColors.text,
              fontSize: 13.5,
              fontWeight: AppFont.black,
            ),
          ),
        if (alt != null)
          Padding(
            padding: const EdgeInsets.only(top: 2, bottom: 8),
            child: Text(
              alt!,
              style: TextStyle(
                color: AppColors.textMuted,
                fontSize: 10.5,
                height: 14 / 10.5,
              ),
            ),
          ),
        ...children,
      ],
    ),
  );
}

class MacRadarBekle extends StatelessWidget {
  const MacRadarBekle({super.key});

  @override
  Widget build(BuildContext context) => Padding(
    padding: EdgeInsets.symmetric(vertical: 12),
    child: Center(child: CircularProgressIndicator(color: AppColors.primary)),
  );
}

/// Veri yokluğu / hata cümlelerinin ortak stili.
// GETTER: dosya düzeyi değişken Dart'ta bir kez hesaplanır ve takım
// teması değişince ESKİ renkte donardı (2026-08-12, emülatörde görüldü).
TextStyle get kMacRadarBos =>
    TextStyle(color: AppColors.textMuted, fontSize: 11.5, height: 16 / 11.5);
